pkg/machine/apple/vfkit-rest.go

```diff
@@ -0,0 +1,59 @@
+//go:build darwin
+
+package apple
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+	"syscall"
+)
+
+// This code is adapted from github.com/crc-org/vfkit/pkg/rest/rest.go as of vkit v0.6.0.
+// We don’t want to import that directly because it imports an enormous dependency tree.
+
+// see `man unix`:
+// UNIX-domain addresses are variable-length filesystem pathnames of at most 104 characters.
+func maxSocketPathLen() int {
+	var sockaddr syscall.RawSockaddrUnix
+	// sockaddr.Path must end with '\0', it's not relevant for go strings
+	return len(sockaddr.Path) - 1
+}
+
+// This is intended to be equivalent to github.com/crc-org/vfkit/pkg/rest.NewEndpoint(input).ToCmdLine()
+func restNewEndpointToCmdLine(input string) ([]string, error) {
+	uri, err := url.ParseRequestURI(input)
+	if err != nil {
+		return nil, err
+	}
+
+	switch strings.ToUpper(uri.Scheme) {
```

Comment:
nit: `uri.Scheme` is always converted to lowercase by net/url (since 2013) so it does not make sense to use ToUpper here.

Reply:
ack, yeah that was how we imported the code so I didn't double check. I will fix this up further to simplify even more then.